### PR TITLE
Refactor: include company name in copyright text

### LINKED ISSUE #1128: [BUG]: Back Arrow on Settings side menu bar not clickable
### Description

On the side nav bar, the user should be able to go back to the previous page.

### Acceptance Criteria

1. The back button when clicked redirects the user to the previous page.
2. The back button on the side navigation bar is fully clickable and functional.
3.  The back button correctly navigates to the previous page or section as intended.
4.  Visual feedback is provided when the back button is hovered over or clicked.
5.  The back button works consistently across different devices, screen sizes, and browsers.
6.  The back button is accessible via keyboard navigation and screen readers.

### Expected Outcome:
The back button on the side navigation bar is fully functional, providing users with an easy and reliable way to navigate back to the previous page or section.

### Links

| [`FIGMA LINK`](https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=24434-47394&t=NFtHhOdRKx2Uw88K-1)
Screenshot 2024-08-22 at 9 28 27 AM
Screenshot 2024-08-23 at 1 25 20 AM

### Images

image

### LINKED ISSUE #1209: [Enhancement]: Add Website Name to "All Rights Reserved" Text in Footer of the landing page
### Description

The footer currently displays "©2024 All Rights Reserved" without the website's name. To enhance brand visibility and ensure proper copyright attribution, the website's name "HNG Boilerplate" should be included in this text.

### Acceptance Criteria

- The website's name is correctly included in the "All Rights Reserved" text in the footer.
- The text is aligned with the rest of the footer content and does not disrupt the layout.

### Links

| [`FIGMA LINK`](LINK_HERE)
https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=2084-18389

### Images

image

### LINKED ISSUE #1233: [Feature]: [FE ANCHOR] Integrate Blog Data Fetching from Backend
### Description

Implement functionality to fetch blog data from the backend and display it on the front end. This will allow the blog section to dynamically display the latest blog posts and related content by retrieving data from the backend API.

### Acceptance Criteria

Data Retrieval:

When the user accesses the blog section, the latest blog posts should be fetched from the backend and displayed dynamically.

If there are multiple blog posts, they should be displayed with pagination.

### Links

[LIVE LINK](https://anchor-python.teams.hng.tech/blog)

##VIDEO DESCRIPTION
(https://github.com/user-attachments/assets/8bd1ac8d-c222-4960-b8f9-52cb87cd86e1)

### PR DESCRIPTION
**Pull request by Feng-09**
# Description
The footer currently displays "©2024 All Rights Reserved" without the website's name. To enhance brand visibility and ensure proper copyright attribution, the website's name "HNG Boilerplate" should be included in this text.

Implement functionality to fetch blog data from the backend and display it on the front end. This will allow the blog section to dynamically display the latest blog posts and related content by retrieving data from the backend API.

**Closes #1209 **
**Closes #1233  **
**Closes #1128  **

# Changes proposed

## What did you do?
I added a companyName property to the footerBottom object in en.json
I used the value of this property in the external pages footer

I consumed the endpoint to fetch blog data from the backend

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [X] My code follows the code style of this project.
- [X] This PR does not contain plagiarized content.
- [X] The title and description of the PR is clear and explains the approach.
- [X] I am making a pull request against the **dev branch** (left side).
- [X] My commit messages styles matches our requested structure.
- [X] My code additions will fail neither code linting checks nor unit test.
- [X] I am only making changes to files I was requested to.

# Screenshots/Videos

### CODE REVIEW

File: src/app/(landing-routes)/blog/[id]/RelatedArticle.tsx [hunk 1]
```diff
@@ -10,29 +11,67 @@ export type New = {
...
+    const fetchData = async () => {
+      try {
+        setIsLoading(true);
+        const response = await fetch(
+          `https://staging.api-csharp.boilerplate.hng.tech/api/v1/blogs`,
+        );
+        const result = await response.json();
+        setData(result.data);
+      } catch {
+        // console.error("Error fetching data:", error);
```
Comment (billmal071) [inline src/app/(landing-routes)/blog/[id]/RelatedArticle.tsx]: Remove

File: src/app/(landing-routes)/blog/[id]/_components/BlogDetailsPage.tsx [hunk 2]
```diff
@@ -38,7 +37,35 @@ const BlogDetailsPage: FC<IProperties> = ({ id }) => {
   const [comments, setComments] =
     useState<Omit<CommentProperties, "session">[]>(sampleComments);
 
-  const post = articlesData.find((item) => item.id === id);
+  // const post = articlesData.find((item) => item.id === id);
```
Comment (billmal071) [inline src/app/(landing-routes)/blog/[id]/_components/BlogDetailsPage.tsx]: Remove

File: src/app/(landing-routes)/blog/[id]/_components/BlogDetailsPage.tsx [hunk 3]
```diff
@@ -38,7 +37,35 @@ const BlogDetailsPage: FC<IProperties> = ({ id }) => {
...
+    const fetchData = async () => {
+      try {
+        setIsLoading(true);
+        const response = await fetch(
+          `https://staging.api-csharp.boilerplate.hng.tech/api/v1/blogs/${id}`,
+        );
+        const result = await response.json();
+        setPost(result.data);
+      } catch {
+        // console.error("Error fetching data:", error);
```
Comment (billmal071) [inline src/app/(landing-routes)/blog/[id]/_components/BlogDetailsPage.tsx]: Remove

File: src/app/(landing-routes)/blog/page.tsx [hunk 4]
```diff
@@ -1,39 +1,87 @@
...
+import {
+  Pagination,
+  PaginationContent,
+  PaginationItem,
+  PaginationLink,
+  PaginationNext,
+  PaginationPrevious,
+} from "~/components/ui/pagination";
+
+// import { blogPosts } from "./data/mock";
```
Comment (billmal071) [inline src/app/(landing-routes)/blog/page.tsx]: Remove

File: src/app/(landing-routes)/blog/page.tsx [hunk 5]
```diff
@@ -1,39 +1,87 @@
...
+    const fetchData = async () => {
+      try {
+        setIsLoading(true);
+        const response = await fetch(
+          `https://staging.api-csharp.boilerplate.hng.tech/api/v1/blogs`,
+        );
+        const result = await response.json();
+        setData(result.data);
+      } catch {
+        // console.error("Error fetching data:", error);
```
Comment (billmal071) [inline src/app/(landing-routes)/blog/page.tsx]: Remove

**Review by billmal071 — DISMISSED** [review]

**Review by Feng-09 — COMMENTED** [review]
I've removed all comments

**Review by prudentbird — APPROVED** [review]

**Review by prudentbird — APPROVED** [review]

**Review by billmal071 — APPROVED** [review]
LGTM

**Pull request merged**